conans/client/client_cache.py

```diff
@@ -4,18 +4,43 @@
 from conans.client.conf import ConanClientConfigParser, default_client_conf, default_settings_yml
 from conans.model.values import Values
 from conans.client.detect import detect_defaults_settings
-from conans.model.ref import ConanFileReference
+from conans.model.ref import ConanFileReference, PackageReference
 from conans.model.manifest import FileTreeManifest
 from conans.paths import SimplePaths, CONANINFO
 from genericpath import isdir
 from conans.model.info import ConanInfo
+from conans.errors import ConanException
 
 
 CONAN_CONF = 'conan.conf'
 CONAN_SETTINGS = "settings.yml"
 LOCALDB = ".conan.db"
 REGISTRY = "registry.txt"
 PROFILES_FOLDER = "profiles"
+_CACHED_BUILD_ID = "_conn_b_id"  # no collision with existing user vars
+
+
+def has_build_id(conanfile):
+    return hasattr(conanfile, _CACHED_BUILD_ID)
+
+
+def build_id(conanfile):
```

Comment:
Maybe `has_build_id` and `build_id` (with different name) should be  conanfile methods?  Too much logic for client_cache